taskcollector/TaskCollector.cpp

```diff
@@ -1,81 +1,78 @@
 #include "FreeImage.h"
+#include <algorithm>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <list>
+#include <memory>
 #include <queue>
 #include <iostream>
 
-
-std::queue<FIBITMAP *> alphaChunks;	
+using bitmap_t = std::shared_ptr<FIBITMAP>;
 
 /** Generic image loader
-	@param lpszPathName Pointer to the full file name
-	@param flag Optional load flag constant
-	@return Returns the loaded dib if successful, returns NULL otherwise
+@param lpszPathName Pointer to the full file name
+@param flag Optional load flag constant
+@return Returns the loaded dib if successful, returns NULL otherwise
 */
-FIBITMAP* GenericLoader(const char* lpszPathName, int flag) {
+bitmap_t GenericLoader(const char* lpszPathName, int flag) {
 	FREE_IMAGE_FORMAT fif = FIF_UNKNOWN;
 
 	// check the file signature and deduce its format
 	// (the second argument is currently not used by FreeImage)
 	fif = FreeImage_GetFileType(lpszPathName, 0);
-	if(fif == FIF_UNKNOWN) {
+	if (fif == FIF_UNKNOWN) {
 		// no signature ?
 		// try to guess the file format from the file extension
 		fif = FreeImage_GetFIFFromFilename(lpszPathName);
 	}
 	// check that the plugin has reading capabilities ...
-	if((fif != FIF_UNKNOWN) && FreeImage_FIFSupportsReading(fif)) {
+	if ((fif != FIF_UNKNOWN) && FreeImage_FIFSupportsReading(fif)) {
 		// ok, let's load the file
-		FIBITMAP *dib = FreeImage_Load(fif, lpszPathName, flag);
+		auto dib = std::make_shared<FIBITMAP>(*FreeImage_Load(fif, lpszPathName, flag));

```

Comment:
I understand what you wanted to get here, but it is done incorrectly.

Shared pointer will just free the memory by `delete` operator. There is no guarantee that the memory was allocated in compatible way. You should always put resource back to libraries they come from.  FreeImage library expects the function `FreeImage_Unload()` to be called on `FIBITMAP*`.

To make it correct you can use create RAII wrapper for `FIBITMAP*` or use `std::unique_ptr` with a custom deleter that calls `FreeImage_Unload()`.